apps/meteor/server/models/raw/Subscriptions.ts

```diff
@@ -603,6 +603,21 @@ export class SubscriptionsRaw extends BaseRaw<ISubscription> implements ISubscri
 		return this.updateOne(query, update);
 	}
 
+	updateAllAutoTranslateLanguagesByUserId(userId: IUser['_id'], language: string): Promise<UpdateResult | Document> {
+		const query = {
+			'u._id': userId,
+			'autoTranslate': true,
```

Comment:
I would like to ask, what happens if I set a different language for a specific room? Will I lose my configuration? Are we heading in a direction where language (based on user preference) is the way to go? If so, I wouldn't save the language in the subscription anymore, but use the user's preference to determine the language.

Reply:
Yes, whenever you update your user's language, all the subscription's `autoTranslateLanguage` would be set to be the new user's language. I guess using the user's preference would be better, but I am not sure whether the feature to set a specific language for a specific channel is altogether not useful to keep, and I believe it would be a lot more complicated to have some kind of hierarchy of languages to choose instead of just using the Subscription value always.

I guess we could do something like this, to define such hierarchy:
```
if subscription.autoTranslate && subscription.autoTranslateLanguage
  translateMessages(subscription.autoTranslateLanguage)
else
  translateMessages(user.language)
```

But I am not sure if this would be preferable, we could also have a setting for this, but I am not sure, what do you think @matheusbsilva137

Reply:
> I would like to ask, what happens if I set a different language for a specific room? Will I lose my configuration? 

Yes, you would. That's the behavior we planned for this case.

> Are we heading in a direction where language (based on user preference) is the way to go?

Not exactly. That will only be true when the new `AutoTranslate_AutoEnableOnJoinRoom` setting is enabled. This setting has been introduced recently, and its idea is to turn on auto-translate automatically when a user who has a language preference different from the workspace's default joins a room. We're introducing this new behavior now so that the auto-translate language is always synced with the language preference **when this setting is enabled**.

Reply:
@gustrb liked your approach, that'd allow users to set custom translation preferences on rooms and when that's not defined we'd just use their language preference.

Still, generally speaking, I'm not sure it's really worth changing the approach we already have since I can't think of any practical benefits :thinking: When the language preference is changed, we'd still need to run through the user's subscription (in this case, to remove the `autoTranslateLanguage` field) AND we'd even need to add a preference "query" to check their language.

In my perspective, since this is a specific case for when a setting is turned on, I don't believe it's worth creating a whole new flow for that (also, I can't see any performance improvements on it). Would like to hear @ggazzo thoughts about it, though.

Reply:
IMO the best solution would be:
1. Have a user's preference for the language
2. Have a subscription's preference for the language (which by default is "inherited" from the user's preference, or in other words, it says to use the user's preference setting
3. The check for the language to use when processing the translation checks the subscription setting, it not set or set to inherit it uses the user's setting
4. A button to reset all subscription defined language to "inherited"

That way we prevent an update on many records if the user changes his language, which can be heavy and can happen by mistake.

What do you guys think?

Reply:
In think it is a good approach. As we talked earlier today, let's bring these improvements to another task.